pyrealm/core/solar.py

```diff
@@ -1,924 +1,1066 @@
 """The :mod:`~pyrealm.core.solar` submodule provides functions to calculate
-photosynthetic photon flux density (ppfd), daily solar radiation fluxes and other
+photosynthetic photon flux density (PPFD), daily solar radiation fluxes and other
 radiative values.
+
+Several of the calculations share a set of intermediate values (:math:`r_u, r_v, r_w`),
+which are calculated using :meth:`calculate_ru_rv_intermediates` and
+:meth:`calculate_rw_intermediate`. The functions for these calculations have a public
+function that uses standard input variables, and a second private version of the
+function that accepts precalculated values of :math:`r_u, r_v, r_w`. These private
+functions can be used to avoid repeated calculation of intermediate values where
+multiple functions need to be run.
 """  # noqa: D205
 
 import numpy as np
 from numpy.typing import NDArray
 
 from pyrealm.constants import CoreConst
-from pyrealm.core.calendar import LocationDateTime
 from pyrealm.core.utilities import check_input_shapes
 
 
-def calc_distance_factor(nu: NDArray, k_e: float) -> NDArray:
+def calculate_distance_factor(
+    nu: NDArray[np.float64], solar_eccentricity: float = CoreConst().solar_eccentricity
+) -> NDArray[np.float64]:
     r"""Calculates distance factor.
 
-    This function calculates the distance factor :math:`dr` using the method of
+    This function calculates the distance factor :math:`d_r` using the method of
     :cite:t:`berger:1993a`.
 
     .. math::
 
-        dr = \left( 1.0 \mathbin{/}
-               \left(\frac{1.0 - k_{e}^2}
-                          {1.0 + k_{e} \cos\left(\nu \cdot \pi \mathbin{/}
-                          180)\right)}
+        d_r = \left( 1.0 \mathbin{/}
+               \left(\frac{1.0 - e^2}
+                          {1.0 + e \cos\left(\nu) \right)}
                \right)
              \right)^2
 
     Args:
-        nu: Heliocentric true anomaly (degrees)
-        k_e: Solar eccentricity
+        nu: Heliocentric true anomaly (:math:`\nu`, degrees)
+        solar_eccentricity: Solar eccentricity (:math:`e`), defaulting to
```

Comment:
When I read `e` in the equation above I automatically assumed Euler's number, so maybe a less misleading name would be better.

Reply:
Unfortunately that is the standard notation: https://en.wikipedia.org/wiki/Orbital_eccentricity